android/jni/com/mapswithme/maps/Framework.cpp

```diff
@@ -918,29 +920,20 @@ Java_com_mapswithme_maps_Framework_nativeGetDistanceAndAzimuthFromLatLon(
 }
 
 JNIEXPORT jobject JNICALL
-Java_com_mapswithme_maps_Framework_nativeFormatLatLon(JNIEnv * env, jclass, jdouble lat, jdouble lon, jboolean useDMSFormat)
-{
-  return jni::ToJavaString(
-      env, (useDMSFormat ? measurement_utils::FormatLatLonAsDMS(lat, lon, 2)
-                         : measurement_utils::FormatLatLon(lat, lon, true /* withSemicolon */, 6)));
-}
-
-JNIEXPORT jobjectArray JNICALL
-Java_com_mapswithme_maps_Framework_nativeFormatLatLonToArr(JNIEnv * env, jclass, jdouble lat, jdouble lon, jboolean useDMSFormat)
+Java_com_mapswithme_maps_Framework_nativeFormatLatLon(JNIEnv * env, jclass, jdouble lat, jdouble lon, int coordsFormat)
 {
-  string slat, slon;
-  if (useDMSFormat)
-    measurement_utils::FormatLatLonAsDMS(lat, lon, slat, slon, 2);
-  else
-    measurement_utils::FormatLatLon(lat, lon, slat, slon, 6);
-
-  static jclass const klass = jni::GetGlobalClassRef(env, "java/lang/String");
-  jobjectArray arr = env->NewObjectArray(2, klass, 0);
-
-  env->SetObjectArrayElement(arr, 0, jni::ToJavaString(env, slat));
-  env->SetObjectArrayElement(arr, 1, jni::ToJavaString(env, slon));
-
-  return arr;
+  switch (coordsFormat) // See CoordinatesFormat enum for all possible values.
+  {
+    default:
+    case 0: // DMS, comma separated
+      return jni::ToJavaString(env, measurement_utils::FormatLatLonAsDMS(lat, lon, true /*withComma*/, 2));
+    case 1: // Decimal, comma separated
+      return jni::ToJavaString(env, measurement_utils::FormatLatLon(lat, lon, true /* withComma */, 6));
+    case 2: // Open location code, long format
+      return jni::ToJavaString(env, openlocationcode::Encode({lat, lon}));
+    case 3: // Link to osm.org
+      return jni::ToJavaString(env, measurement_utils::FormatOsmLink(lat, lon, 14));
+  }
```

Comment:
I'd prefer introduce local cpp enum named "CoordinatesFormat" in this module at least to write comment that it should be equal with Java enum CoordinatesFormat.

Reply:
Introduced `CoordinatesFormat` in C++ code to reflect Java enum